k8s/scripts/inspect.sh

```diff
@@ -0,0 +1,169 @@
+
+function log_info {
+  printf -- '\033[34m INFO: \033[0m %s\n' "$1"
+}
+
+function log_warning() {
+  printf -- '\033[33m WARNING: \033[0m %s\n' "$1"
+}
+
+function log_warning_red {
+  printf -- '\033[31m WARNING: \033[0m %s\n' "$1"
+}
+
+function is_control_plane_node {
+  k8s local-node-status | grep -q "control-plane"
+}
+
+function is_service_active {
+  local service
+  service=$1
+
+  systemctl status "snap.$service" | grep -q "active (running)"
+}
+
+function collect_args {
+  log_info "Copy service args to the final report tarball"
+  cp -r --no-preserve=mode,ownership /var/snap/k8s/common/args "$INSPECT_DUMP"
+}
+
+function collect_cluster_info {
+  log_info "Copy k8s cluster-info dump to the final report tarball"
+  k8s kubectl cluster-info dump --output-directory "$INSPECT_DUMP/cluster-info" &>/dev/null
+}
+
+function collect_sbom {
+  log_info "Copy SBOM to the final report tarball"
+  cp --no-preserve=mode,ownership /snap/k8s/current/bom.json "$INSPECT_DUMP"/sbom.json
+}
+
+function collect_k8s_diagnostics {
+  log_info "Copy uname to the final report tarball"
+  uname -a &>"$INSPECT_DUMP/uname.log"
+
+  log_info "Copy snap diagnostics to the final report tarball"
+  snap version &>"$INSPECT_DUMP/snap-version.log"
+  snap list k8s &>"$INSPECT_DUMP/snap-list-k8s.log"
+  snap services k8s &>"$INSPECT_DUMP/snap-services-k8s.log"
+  snap logs k8s -n 10000 &>"$INSPECT_DUMP/snap-logs-k8s.log"
+
+  log_info "Copy k8s diagnostics to the final report tarball"
+  k8s kubectl version &>"$INSPECT_DUMP/k8s-version.log"
+  k8s status &>"$INSPECT_DUMP/k8s-status.log"
+  k8s get &>"$INSPECT_DUMP/k8s-get.log"
+  k8s kubectl get cm k8sd-config -n kube-system -o yaml &>"$INSPECT_DUMP/k8s.k8sd/k8sd-configmap.log"
+  k8s kubectl get cm -n kube-system &>"$INSPECT_DUMP/k8s-configmaps.log"
+
+  cp --no-preserve=mode,ownership /var/snap/k8s/common/var/lib/k8s-dqlite/cluster.yaml "$INSPECT_DUMP/k8s.k8s-dqlite/k8s-dqlite-cluster.yaml"
+  cp --no-preserve=mode,ownership /var/snap/k8s/common/var/lib/k8s-dqlite/info.yaml "$INSPECT_DUMP/k8s.k8s-dqlite/k8s-dqlite-info.yaml"
+  cp --no-preserve=mode,ownership /var/snap/k8s/common/var/lib/k8sd/state/database/cluster.yaml "$INSPECT_DUMP/k8s.k8sd/k8sd-cluster.yaml"
+  cp --no-preserve=mode,ownership /var/snap/k8s/common/var/lib/k8sd/state/database/info.yaml "$INSPECT_DUMP/k8s.k8sd/k8sd-info.yaml"
+
+  ls -la /var/snap/k8s/common/var/lib/k8s-dqlite &>"$INSPECT_DUMP/k8s.k8s-dqlite/k8s-dqlite-files.log"
+  ls -la /var/snap/k8s/common/var/lib/k8sd &>"$INSPECT_DUMP/k8s.k8sd/k8sd-files.log"
+}
+
+function collect_service_diagnostics {
+  local service
+  service=$1
+
+  mkdir -p "$INSPECT_DUMP/$service"
+
+  local status_file
+  status_file="$INSPECT_DUMP/$service/systemctl.log"
+
+  systemctl status "snap.$service" &>"$status_file"
+
+  local n_restarts
+  n_restarts=$(systemctl show "snap.$service" -p NRestarts | cut -d'=' -f2) 
+
+  printf -- "%s NRestarts=%s" "$service" "$n_restarts" >> "$INSPECT_DUMP/nrestarts.log"
```

Comment:
nit: the file is already named `nrestarts.log`.
Let's just have something like 
```
$service -> $s restarts
```
or even 
```
$service -> $s
```